deno/main.ts

```diff
@@ -0,0 +1,70 @@
+// Import types using skypack for deno support
+import { Request } from "https://cdn.skypack.dev/@voiceflow/base-types";
+
+const API_KEY = "YOUR_API_KEY_HERE"; // it should look like this: VF.XXXXXXX.XXXXXX... keep this a secret!
+const VERSION_ID = "YOUR_VERSION_ID_HERE"; // your Voiceflow project versionID, find it under the "integrations" tab
+
+const API_URL = `https://general-runtime.voiceflow.com/`;
+
+// send an interaction to the Voiceflow API, and log the response, returns true if there is a next step
+async function interact<R extends typeof Request.BaseRequest>(
+  userID: string,
+  request: R,
+): Promise<boolean> {
+  const url = new URL(`state/${VERSION_ID}/user/${userID}/interact`, API_URL);
+
+  // call the Voiceflow API with the user's name & request, get back a response
+  const response = await fetch(url, {
+    method: "POST",
+    headers: {
+      Authorization: API_KEY,
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({request}),
+  });
+
+  if (!response.ok) {
+    console.log("Uh oh", response.status, response.statusText);
+    return false;
+  }
+
+  const json = await response.json();
+  console.log('response', json);
+  // loop through the response
+  for (const trace of json) {
+    switch (trace.type) {
+      case "text":
+      case "speak": {
+        console.log(trace.payload.message);
+        break;
+      }
+      case "end": {
+        // an end trace means the the Voiceflow dialog has ended
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+const userID = prompt("> What is your name?");
```

Comment:
what do u think if we encapsulate this end part into a `main` function and call it at the end? like this:

```ts
async function main() {
  const userID = prompt("> What is your name?");

  if (userID) {
    // send a simple launch request starting the dialog
    let isRunning = await interact<typeof Request.LaunchRequest>(userID, {
      type: Request.RequestType.LAUNCH,
    });
  
    while (isRunning) {
      const nextInput = prompt("> Say something");
      if (!nextInput) break;
  
      // send a simple text type request with the user input
      isRunning = await interact<typeof Request.TextRequest>(userID, {
        type: Request.RequestType.TEXT,
        payload: nextInput,
      });
    }
  }
  
  console.log("The end!");
}

// Runs the app
main();
```

Reply:
What benefit do you see with this change? Deno supports top level await so we aren't required to nest it in an async function.

Reply:
just a matter of standard and organization since the nodejs example have something like this I guess, but it's up to u. Not a strong opinion here